apps/examples/ble_rmc/ble_rmc_main.c

```diff
@@ -339,29 +365,43 @@ int ble_rmc_main(int argc, char *argv[])
 		}
 	}
 
+	if (strncmp(argv[1], "disconn", 8) == 0) {
+		ret = ble_client_disconnect(g_ctx);
+		if (ret != BLE_MANAGER_SUCCESS) {
+			RMC_LOG(RMC_TAG, "disconnect fail[%d]\n", ret);
+			return 0;
+		}
+		RMC_LOG(RMC_TAG, "disconnect ok\n");
+	}
+
 	if (strncmp(argv[1], "connect", 8) == 0) {
 		struct timespec abstime;
+		ble_client_state_e cli_state = BLE_CLIENT_NONE;
 
-		ble_scan_filter filter = { 0, };
-		memcpy(&(filter.raw_data), ble_filter, sizeof(ble_filter));
-		filter.raw_data_length = sizeof(ble_filter);
-		filter.scan_duration = 1000;
-		g_scan_done = 0;
-		ret = ble_client_start_scan(&filter);
+		if (argc == 3 && strncmp(argv[2], "fail", 5) == 0) {
+			memset(g_target, 1, BLE_BD_ADDR_MAX_LEN);
+		} else {
+			ble_scan_filter filter = { 0, };
+			memcpy(&(filter.raw_data), ble_filter, sizeof(ble_filter));
+			filter.raw_data_length = sizeof(ble_filter);
+			filter.scan_duration = 1000;
+			g_scan_done = 0;
+			ret = ble_client_start_scan(&filter, &scan_config);
 
-		if (ret != BLE_MANAGER_SUCCESS) {
-			RMC_LOG(RMC_TAG, "scan start fail[%d]\n", ret);
-			return 0;
-		}
+			if (ret != BLE_MANAGER_SUCCESS) {
+				RMC_LOG(RMC_TAG, "scan start fail[%d]\n", ret);
+				return 0;
+			}
 
-		sleep(2);
+			sleep(2);
 
-		if (g_scan_done != 1) {
-			RMC_LOG(RMC_TAG, "scan fail to get adv packet\n", ret);
-			return 0;
+			if (g_scan_done != 1) {
+				RMC_LOG(RMC_TAG, "scan fail to get adv packet\n", ret);
```

Comment:
`ret` is not used anywhere. Is it needed?

Reply:
I fixed it